jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/MavenLayerConfigurationsTest.java

```diff
@@ -315,4 +298,106 @@ private Artifact makeArtifact(Path path) {
+    ImmutableList<Path> expectedResourcesFiles =
+        ImmutableList.of(
+            outputPath.resolve("final-name/META-INF/context.xml"),
+            outputPath.resolve("final-name/Test.jsp"),
+            outputPath.resolve("final-name/WEB-INF/classes/empty_dir"),
+            outputPath.resolve("final-name/WEB-INF/classes/package/test.properties"),
+            outputPath.resolve("final-name/WEB-INF/web.xml"));
+    ImmutableList<Path> expectedClassesFiles =
+        ImmutableList.of(
+            outputPath.resolve("final-name/WEB-INF/classes/HelloWorld.class"),
+            outputPath.resolve("final-name/WEB-INF/classes/empty_dir"), // Not sure about that
+            outputPath.resolve("final-name/WEB-INF/classes/package/Other.class"));
+
+    assertSourcePathsUnordered(
+        expectedDependenciesFiles, configuration.getDependencyLayerEntries());
+    assertSourcePathsUnordered(
+        expectedSnapshotDependenciesFiles, configuration.getSnapshotDependencyLayerEntries());
+    assertSourcePathsUnordered(expectedResourcesFiles, configuration.getResourceLayerEntries());
+    assertSourcePathsUnordered(expectedClassesFiles, configuration.getClassLayerEntries());
+
+    assertExtractionPathsUnordered(
+        Arrays.asList("/my/app/WEB-INF/lib/dependency-1.0.0.jar"),
+        configuration.getDependencyLayerEntries());
+    assertExtractionPathsUnordered(
+        Arrays.asList("/my/app/WEB-INF/lib/dependencyX-1.0.0-SNAPSHOT.jar"),
+        configuration.getSnapshotDependencyLayerEntries());
+    assertExtractionPathsUnordered(
+        Arrays.asList(
+            "/my/app/META-INF/context.xml",
+            "/my/app/Test.jsp",
+            "/my/app/WEB-INF/classes/empty_dir",
+            "/my/app/WEB-INF/classes/package/test.properties",
+            "/my/app/WEB-INF/web.xml"),
+        configuration.getResourceLayerEntries());
+    assertExtractionPathsUnordered(
+        Arrays.asList(
+            "/my/app/WEB-INF/classes/HelloWorld.class",
+            "/my/app/WEB-INF/classes/empty_dir", // Not sure about that
+            "/my/app/WEB-INF/classes/package/Other.class"),
+        configuration.getClassLayerEntries());
+    assertExtractionPathsUnordered(
+        Arrays.asList("/a/b/bar", "/c/cat", "/foo"), configuration.getExtraFilesLayerEntries());
+  }
+
+  @Test
+  public void testGetForJarProject_nonDefaultAppRoot() throws URISyntaxException, IOException {
+    Path extraFilesDirectory = Paths.get(Resources.getResource("layer").toURI());
+    // Test when the default packaging is set
+    Mockito.when(mockMavenProject.getPackaging()).thenReturn("jar");
+
+    AbsoluteUnixPath appRoot = AbsoluteUnixPath.get("/my/app");
+    JavaLayerConfigurations configuration =
+        MavenLayerConfigurations.getForProject(mockMavenProject, extraFilesDirectory, appRoot);
+
+    assert_nonDefaultAppRoot(configuration);
+  }
+
+  private void assert_nonDefaultAppRoot(JavaLayerConfigurations configuration) {
```

Comment:
Can this be `private static`? If so, change it to `static` and move the method code above any non-static methods or fields.

Reply:
Done